Userland/Utilities/netstat.cpp

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright (c) 2021, Brandon Pruitt <brapru@pm.me>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/JsonArray.h>
+#include <AK/JsonObject.h>
+#include <AK/QuickSort.h>
+#include <AK/String.h>
+#include <LibCore/ArgsParser.h>
+#include <LibCore/File.h>
+#include <unistd.h>
+
+int main(int argc, char** argv)
+{
+    if (pledge("stdio rpath", nullptr) < 0) {
+        perror("pledge");
+        return 1;
+    }
+
+    if (unveil("/proc/net", "r") < 0) {
+        perror("unveil");
+        return 1;
+    }
+
+    if (unveil(nullptr, nullptr) < 0) {
+        perror("unveil");
+        return 1;
+    }
+
+    bool flag_all = false;
+    bool flag_list = false;
+    bool flag_tcp = false;
+    bool flag_udp = false;
+
+    Core::ArgsParser args_parser;
+    args_parser.set_general_help("Display network connections");
+    args_parser.add_option(flag_all, "Display both listening and non-listening sockets", "all", 'a');
+    args_parser.add_option(flag_list, "Display only listening sockets", "list", 'l');
+    args_parser.add_option(flag_tcp, "Display only TCP network connections", "tcp", 't');
+    args_parser.add_option(flag_udp, "Display only UDP network connections", "udp", 'u');
+    args_parser.parse(argc, argv);
+
+    bool has_protocol_flag = (flag_tcp || flag_udp);
+
+    enum class Alignment {
+        Left,
+        Right
+    };
+
+    struct Column {
+        String title;
+        Alignment alignment { Alignment::Left };
+        int width { 0 };
+        String buffer;
+    };
+
+    Vector<Column> columns;
+
+    int protocol_column = -1;
+    int bytes_in_column = -1;
+    int bytes_out_column = -1;
+    int local_address_column = -1;
+    int peer_address_column = -1;
+    int state_column = -1;
+
+    auto add_column = [&](auto title, auto alignment, auto width) {
+        columns.append({ title, alignment, width, {} });
+        return columns.size() - 1;
+    };
+
+    protocol_column = add_column("Proto", Alignment::Left, 5);
+    bytes_in_column = add_column("Bytes-In", Alignment::Right, 9);
+    bytes_out_column = add_column("Bytes-Out", Alignment::Right, 9);
+    local_address_column = add_column("Local Address", Alignment::Left, 20);
+    peer_address_column = add_column("Peer Address", Alignment::Left, 20);
+    state_column = add_column("State", Alignment::Left, 9);
+
+    auto print_column = [](auto& column, auto& string) {
+        if (!column.width) {
+            out("{}", string);
+            return;
+        }
+        if (column.alignment == Alignment::Right) {
+            out("{:>{1}}  "sv, string, column.width);
+        } else {
+            out("{:<{1}}  "sv, string, column.width);
+        }
+    };
+
+    if (!has_protocol_flag || flag_tcp || flag_udp) {
+        out("Active Internet connections ");
+
+        if (flag_all)
+            outln("(servers and established)");
+        else {
```

Comment:
```suggestion
        if (flag_all) {
            outln("(servers and established)");
        } else {
```

Mixing non-braced if and braced else looks weird.

Reply:
Fixed 👍